### PR TITLE
Default-enable `llvm_tools_enabled` when no `config.toml` is present

### LINKED ISSUE #133195: Bootstrap without creating a config.toml fails with missing rust-objcopy
After cloning, I expected `python3 x.py test --stage 1 tests/ui` to succeed but instead I got many test failures like this one:

```
---- [ui] tests/ui/async-await/track-caller/panic-track-caller.rs#afn stdout ----

error in revision `afn`: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/Users/wesley/Code/rust/build/aarch64-apple-darwin/stage1/bin/rustc" "/Users/wesley/Code/rust/tests/ui/async-await/track-caller/panic-track-caller.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/wesley/.cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/Users/wesley/Code/rust/vendor" "--sysroot" "/Users/wesley/Code/rust/build/aarch64-apple-darwin/stage1" "--target=aarch64-apple-darwin" "--cfg" "afn" "--check-cfg" "cfg(FALSE,afn,cls,nofeat)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/Users/wesley/Code/rust/build/aarch64-apple-darwin/test/ui/async-await/track-caller/panic-track-caller.afn/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/Users/wesley/Code/rust/build/aarch64-apple-darwin/native/rust-test-helpers" "--edition=2021"
stdout: none
--- stderr -------------------------------
error: unable to run `rust-objcopy`: No such file or directory (os error 2)

error: aborting due to 1 previous error
------------------------------------------
```

However, if I create a `config.toml` by `cp config.example.toml config.toml`, then the problem goes away.

I instrumented the code here with a `println!()` and I can see that when `config.toml` exists, this block runs but when `config.toml` does not exist, it does not run:

https://github.com/rust-lang/rust/blob/b71fb5edc0217eaf8fc824a44cd7b0945e29ff4d/src/bootstrap/src/core/build_steps/compile.rs#L1980-L1988

Possibly related to #132720 cc @jieyouxu 

### Meta

Cloned rust-lang/rust at b71fb5edc0217eaf8fc824a44cd7b0945e29ff4d

Host is `aarch64-apple-darwin`

### PR DESCRIPTION
**Pull request by jieyouxu**
Fixes #133195. cc @wesleywiser could you double check if with this patch and no `config.toml` that you can run `./x test tests/ui --stage 1`?

`llvm-objcopy` is usually required by cg_ssa on macOS to workaround bad `strip`s.

cc @bjorn3 I hope this doesn't break cg_clif...

r? bootstrap

### CODE REVIEW

**Comment by bjorn3** [comment]
> I hope this doesn't break cg_clif...

Does this disable the option? https://github.com/rust-lang/rust/blob/e6c1e14e5d487a90abeb5ba4dccc608521bbfb54/compiler/rustc_codegen_cranelift/scripts/setup_rust_fork.sh#L32-L33 If not, it either should, or the config file there will need to be changed to explicitly disable it.

File: src/bootstrap/src/core/config/config.rs [hunk 1]
```diff
@@ -1255,6 +1255,10 @@ impl Config {
             },
             out: PathBuf::from("build"),
 
+            // This is needed by codegen_ssa on macOS to ship `llvm-objcopy` aliased to
+            // `rust-objcopy` to workaround bad `strip`s on macOS.
+            llvm_tools_enabled: true,
```
Comment (Kobzol) [inline src/bootstrap/src/core/config/config.rs:1260]: Funny, `config.example.toml` already claimed that it was enabled by default before.

File: src/bootstrap/src/utils/change_tracker.rs [hunk 2]
```diff
@@ -300,4 +300,9 @@ pub const CONFIG_CHANGE_HISTORY: &[ChangeInfo] = &[
         severity: ChangeSeverity::Info,
         summary: "`download-rustc='if-unchanged'` is now a default option for library profile.",
     },
+    ChangeInfo {
+        change_id: 133207,
+        severity: ChangeSeverity::Info,
+        summary: "llvm tools are is now enabled by default when no `config.toml` is provided.",
```
Comment (Kobzol) [inline src/bootstrap/src/utils/change_tracker.rs:306]:
```suggestion
        summary: "`rust.llvm-tools` is now enabled by default when no `config.toml` is provided.",
```

**Comment by jieyouxu** [comment]
> If not, it either should, or the config file there will need to be changed to explicitly disable it.

I modified the config there to explicitly disable llvm tools.

File: src/bootstrap/src/utils/change_tracker.rs (hunk 2 above)
Comment (jieyouxu) [reply src/bootstrap/src/utils/change_tracker.rs]: Fixed.

**Comment by jieyouxu** [comment]
Changes since last review ([diff](https://github.com/rust-lang/rust/compare/2c39c2bf97573e7cb17ed7de2e1e4a205ee46a3b..70592623a1f527ed1cb422cec8b3ea3810593fbc)):

- Fixed change log entry
- Adjusted cg_clif's config.toml to disable llvm tools

File: compiler/rustc_codegen_cranelift/scripts/setup_rust_fork.sh [hunk 3]
```diff
@@ -38,6 +38,9 @@ local-rebuild = true
 codegen-backends = ["cranelift"]
 deny-warnings = false
 verbose-tests = false
+# Enabled by default, but cranelift does not need/want llvm tools.
```
[the remark this answers is not in the document]
Comment (jieyouxu) [reply compiler/rustc_codegen_cranelift/scripts/setup_rust_fork.sh]: Fixed comment

**Comment by Kobzol** [comment]
LGTM, if bjorn3 is ok with the latest change.

**Review by bjorn3 — APPROVED** [review]

**Review by wesleywiser — APPROVED** [review]
I can confirm this fixes building without `config.toml` for me. Thanks @jieyouxu! 💛

**Pull request merged**